.github/workflows/on-tag.yml

```diff
@@ -0,0 +1,52 @@
+name: Run Docker buildx
+env:
+        DOCKER_CLI_EXPERIMENTAL: enabled
+        TAG_FMT: '^refs/tags/(((.?[0-9]+){3,4}))$'
+
+on:
+        push:
+                tags: [ '*' ]
+
+jobs:
+        build:
+                runs-on: ubuntu-18.04
+                name: Build umbrel middleware
+                steps:
+                        - name: Show github ref
+                          run: echo $GITHUB_REF
+                        - name: setup environment
+                          run: |
+                                  if ! echo "$GITHUB_REF" | grep -qE "$TAG_FMT"; then
+                                        echo "ERR: TAG must be in format: vX.Y.Z or X.Y.Z or vW.X.Y.Z or W.X.Y.Z"
+                                        exit 1
+                                  fi
+                                  VERSION="$(echo "$GITHUB_REF" | sed -E "s|$TAG_FMT|\2|")"
+                                  
+                                  TAG="$(echo "$GITHUB_REF" | sed -E "s|$TAG_FMT|\1|")"
+                                  echo ::set-env name=TAG::"$TAG"
+                                  
+                        - name: Debug
+                          run: |
+                                  printf "    TAG: %s\n"  "$TAG"                                  
+                        - name: Login to Docker for building
+                          run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "nolim1t" --password-stdin
```

Comment:
It would be great if you can put the dockerhub username in secrets too. That way our individual forks will pick up our docker hub usernames from the fork's repo settings, plus generally a good OPSEC

Reply:
good idea